docs/pages/tutorials/embedded-mode-with-nextjs-frontend.mdx

```diff
@@ -0,0 +1,271 @@
+import { Markdown } from '../../components/Page';
+
+# How to embed Keystone in a NextJS app
+
+Keystone now supports SQLite! 🎉
+
+This makes it possible to run a local instance of Keystone from the same place you keep your frontend code and commit everything to Git. If you're happy to apply content changes in a local-only dev environment, and don't need a writeable API in production this use case may be handy for you.
+
+---
+
+[Keystone Lite](https://github.com/keystonejs/keystone/tree/master/examples/next-lite) is a streamlined [mode](/guides/modes) that you can operate Keystone with. It takes advantage of SQLite’s embedded local storage capabilities to give you a powerful CMS that you can run locally and store content in the same place you keep your frontend code.
+
+The tutorial will show you how to: 
+
+- Setup a NextJS app.
+- Embed Keystone within it, and run an Admin UI you can read and write to locally.
+- Create a simple Keystone schema with a `Post` [list](/apis/schema) that NextJS can build pages with. 
+- Setup a secure read-only GraphQL API endpoint (and playground) that you can access in production.
+- Deploy your CMS and frontend monorepo to Vercel. 🚀
+
+?> If you need access to an Keystone’s Admin UI over the internet, or want to provide access to more than one editor, use Keystone in it’s default [standalone](/guides/modes#standalone-mode) mode.
+
+## Setup a NextJS app
+
+Start out by adding a basic NextJS instance empty project directory:
+
+```bash
+yarn create next-app my-project
+cd my-project
+```
+
+### Add TypeScript
+
+Keystone Next is built in TypeScript. Adding it as a `--dev` dependency will make Keystone implementation easier later on. Let‘s get that out of the way now:
+
+```bash
+yarn add --dev typescript @types/react
+```
+Rename the `index.js` file in your `/pages` directory to `index.tsx`. We’ll query data in Keystone using TypeScript later on.
+
+```text
+.
+├── pages
+│   ├── _app.js
+│   ├── api
+│   └── index.tsx // renamed
+```
+
+### Startup your local server
+
+Run `yarn dev` at the root of your project. NextJS will generate a local server for you at [http://localhost:3000](http://localhost:3000).
+
+![A browser showing the Home page of the default NextJS app](/assets/tutorials/next-lite/localhost-home-1.png)
```

Comment:
We kind of skip a step here - this bit shows the default Next.js starter page, down lower (around l.142) we show our own demo homepage (unstyled; pre dynamic content) but we never show the code to paste in and replace the contents of `index.tsx` to get that to render.

I think for clarity / consistency we should add a step either before or after this one that says "replace the contents of your index.tsx page with this snippet"

No strong opinions whether that comes before an updated version of this screenshot, or after (since you're showing the default page in the animated gif lower down)

Could go below here with a "one more thing before we add Keystone... lets add the homepage for our blog, with a hard-coded index of posts for now"